### PR TITLE
[SYCL][PI] Restore CUDA and HIP plugin ABI

### PR DESCRIPTION
**Pull request by kbenzie**
During the port to UR the CUDA and HIP PI plugin ABI's were unintentionally changed. There does not appear to be symbol checks for these plugins, unlike the [Level Zero symbol check](https://github.com/intel/llvm/blob/sycl/sycl/test/abi/pi_level_zero_symbol_check.dump) and [OpenCL symbol check](https://github.com/intel/llvm/blob/sycl/sycl/test/abi/pi_opencl_symbol_check.dump). As such, the ABI change went unnoticed until https://github.com/intel/llvm/pull/10490 was opened using the same approach for the OpenCL port, which [failed](https://github.com/intel/llvm/actions/runs/5610646255/job/15200624025?pr=10490) the OpenCL symbol check.

This PR restores the expected ABI for the CUDA and HIP plugins and introduces new CUDA and HIP symbol check tests.

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
Good catch! Would we be able to add symbol checks for HIP and CUDA or would it make no sense at this point?

**Comment by kbenzie** [comment]
> Good catch! Would we be able to add symbol checks for HIP and CUDA or would it make no sense at this point?

I tried this out locally by copy pasting the Level Zero symbol check, using `REQUIRES: cuda` runs and passes but `REQUIRES: hip` is marked unsupported so I guess that's not an existing lit feature. Also copy pasting is not what the [ABI Policy](https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ABIPolicyGuide.md) suggests.

**Comment by steffenlarsen** [comment]
> > Good catch! Would we be able to add symbol checks for HIP and CUDA or would it make no sense at this point?
> 
> I tried this out locally by copy pasting the Level Zero symbol check, using `REQUIRES: cuda` runs and passes but `REQUIRES: hip` is marked unsupported so I guess that's not an existing lit feature. Also copy pasting is not what the [ABI Policy](https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ABIPolicyGuide.md) suggests.

You should be able to generate the current symbols by using the same `abi_check.py` script and replacing `--mode check_symbols --reference` with `--mode dump_symbols --output`

**Comment by kbenzie** [comment]
> You should be able to generate the current symbols by using the same `abi_check.py` script and replacing `--mode check_symbols --reference` with `--mode dump_symbols --output`

I think the script might need to be slightly modified as it doesn't output a `REQUIRES:` clause and I don't believe plugins and built by default so would probably result in fail test when they are not present.

**Comment by steffenlarsen** [comment]
> > You should be able to generate the current symbols by using the same `abi_check.py` script and replacing `--mode check_symbols --reference` with `--mode dump_symbols --output`
> 
> I think the script might need to be slightly modified as it doesn't output a `REQUIRES:` clause and I don't believe plugins and built by default so would probably result in fail test when they are not present.

Good point. Maybe it can be used to generate the initial version, but then we will have to maintain it without generating it through the script afterwards, like we do for L0 and OpenCL. I am also okay with having a new option for the checker, but I don't know if it makes sense for PI symbols that I assume we will remove eventually anyway.

**Comment by kbenzie** [comment]
> Good point. Maybe it can be used to generate the initial version, but then we will have to maintain it without generating it through the script afterwards, like we do for L0 and OpenCL. I am also okay with having a new option for the checker, but I don't know if it makes sense for PI symbols that I assume we will remove eventually anyway.

I've gone with using the script to generate the checks then manually adding `REQUIRES: cuda_be` and `REQUIRES: hip_be` where appropriate.

**Comment by kbenzie** [comment]
I don't know what's don't wrong with [this check](https://github.com/intel/llvm/actions/runs/5622731453/job/15235925679?pr=10518)...

**Comment by steffenlarsen** [comment]
> I don't know what's don't wrong with [this check](https://github.com/intel/llvm/actions/runs/5622731453/job/15235925679?pr=10518)...

Looks like the Github unicorn strikes again! It was most likely just a github hiccup.

**Review by jchlanda — APPROVED** [review]

File: sycl/plugins/hip/CMakeLists.txt [hunk 1]
```diff
@@ -108,6 +108,7 @@ add_sycl_plugin(hip
     "../unified_runtime/ur/adapters/hip/sampler.cpp"
     "../unified_runtime/ur/adapters/hip/sampler.hpp"
     "../unified_runtime/ur/adapters/hip/usm.cpp"
+    "../unified_runtime/ur/adapters/hip/image.cpp"
```
Comment (jandres742) [inline sycl/plugins/hip/CMakeLists.txt:111]: minor: add new entry on alphabetical order.

**Review by jandres742 — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers this is ready to merge now

**Pull request merged**